pxr/imaging/plugin/hdRpr/python/commonSettings.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2020 Advanced Micro Devices, Inc
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# 
+
+visibility_mask_setting = {
+    'name': 'primvars:rpr:visibilityMask',
+    'ui_name': 'Render Visibility',
+    'defaultValue': '*',
+    'hints': [
+        ('*', 'Visible to all'),
+        ('primary', 'Visible only to primary rays'),
+        ('primary,shadow', 'Visible only to primary and shadow rays'),
+        ('-primary', 'Invisible to primary rays'),
+        ('-primary,-shadow,-light', 'Invisible emissive light'),
+        ('', 'Invisible'),
+    ],
+    'help': 'The visibility mask is a comma-separated list of inclusive or exclusive ray visibility flags.\\n' \
```

Comment:
I think my only question here is only that all are combined to one setting.  Would it make more sense to have different boolean settings for each visibility flag?

Reply:
I thought about implementing it in such a way but a big number of boolean parameters are always very clunky. Imagine that you want to enable just one type of rays, you will need to uncheck all other parameters. It seemed to me like too many clicks.
So I decided to implement it in a very much like in Karma.
If you prefer it to be on the GUI as booleans parameters I can easily change it

Reply:
I think I do.  Mainly because of what we do in other plugins and Renderman, arnold plugins do it this way. At least renderman does.

Reply:
Not to be one to follow the leader but for the simple user this is easier.

Reply:
ok, I will change it